projects/charger/test/test_battery_monitor.c

```diff
@@ -0,0 +1,45 @@
+#include "battery_monitor.h"
+#include "can_transmit.h"
+#include "charger_events.h"
+#include "ms_test_helpers.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+#define TEST_CAN_DEVICE_ID 0x1
+
+static CanStorage s_can_storage;
+static CanSettings s_can_settings = {
+  .device_id = TEST_CAN_DEVICE_ID,         //
+  .bitrate = CAN_HW_BITRATE_250KBPS,       //
+  .tx = { GPIO_PORT_A, 12 },               //
+  .rx = { GPIO_PORT_A, 11 },               //
+  .rx_event = CHARGER_CAN_EVENT_RX,        //
+  .tx_event = CHARGER_CAN_EVENT_TX,        //
+  .fault_event = CHARGER_CAN_EVENT_FAULT,  //
+  .loopback = true                         //
+};
+
+void setup_test(void) {
+  event_queue_init();
+  can_init(&s_can_storage, &s_can_settings);
+  TEST_ASSERT_OK(battery_monitor_init());
```

Comment:
Since `setup_test` is called before every test, every test fails. Since the error generated provides a line number, I'm not concerned that this would make for unclear debugging - the person debugging would know exactly where to look for the failing assertion.

Reply:
Just tested it, assertion failures in `setup_test` are handled gracefully. I think the pattern of putting the init function call in `setup_test` is ok.